interfaces/builtin/opengl.go

```diff
@@ -108,6 +110,12 @@ var openglConnectedPlugUDev = []string{
 	`KERNEL=="vchiq"`,
 }
 
+const openglConnectedPlugSecComp = `
+# Description: allow running operations on GPU devices
+# necessary as cuda tries to create /dev/nvidia-uvm-tools with mknod
+mknod - |S_IFCHR
```

Comment:
The other accesses look ok, but I am not a fan of this rule at all. Granted, to actually create a character device under confinement, you also need 'capability mknod'. Can you explain "necessary as cuda tries to create /dev/nvidia-uvm-tools with mknod" a little bit more? Eg, does this happen only as root, non-root, if the device exists or doesn't exist? What happens if we omit the rule entirely? Do applications fail? (keep in mind, we no longer kill applications with seccomp denials).

Reply:
Indeed. hopefully we don't need this, and if it's absolutely necessary, can we restrict to the specific major/minor for that device?

Reply:
"can we restrict to the specific major/minor for that device?"

Unfortunately no because it is specified via a dev_t structure and can't be dereferenced from userspace (a limitation of seccomp arg filtering). AppArmor could be modified for this, but that isn't on any short/medium term roadmap atm.

Reply:
I'll try to work with the PR locally. A remedy would be to expect the user to create the actual device node with the right permissions before the application is started. I don't think it's a huge issue and seems to be in line with how older versions of CUDA operated, i.e. the node had to be created manually once `nvidia-uvm` got modprobe'd.

Reply:
Loading the driver and creating the node manually before starting apps using CUDA seems to work. I was able to run some of the tests and run some samples. Although I didn't manage to reach the end of the test suite as it seemed to be getting stuck on my GT1030.

I'll drop this seccomp block and push a patch.